src/lexer.rs

```diff
@@ -139,37 +135,46 @@ impl<'a> Iterator for ReplaceComments<'a> {
     type Item = CharAndLocation;
 
     fn next(&mut self) -> Option<Self::Item> {
-        let current = self.inner.next()?;
+        let mut current = self.inner.next()?;
 
         if current.0 != '/' {
-            assert!(current.0 != COMMENT_SENTINEL_VALUE);
+            debug_assert!(current.0 != COMMENT_SENTINEL_VALUE);
             return Some(current);
         }
 
-        let mut save_point = self.inner;
+        let mut save_point = self.inner.clone();
         match self.next() {
             // The // case, consume until but not including the next \n
             Some(('/', _)) => {
-                save_point = self.inner;
-                while let Some((next, _)) = self.inner.next() {
+                current.1.end += 1;
```

Comment:
Instead of manually counting start and end like this, I wonder if we could do something where we can ask the CharsAndLocations for it's current offset compared to the original string with a method like: https://stackoverflow.com/questions/50781561/how-to-find-the-starting-offset-of-a-string-slice-of-another-string

Lines need to be counted manually, but the `Chars` iterator / `&str` shows where we are in the string.

Reply:
I've noticed a problem as `ReplaceComments` is beyond a `Peekable` it's impossible to call any methods on it :(

Reply:
Ok, we can land as is and I'll give it a try as well.

Reply:
If you want I can share the code I have in a different branch

Reply:
Please

Reply:
Did you make any progress or can we merge it has is?

Reply:
Oh sorry, I was thinking we can merge and I'll take a look asynchronously (though there's a merge conflict now)

Reply:
Didn't notice there was a merge conflict